### ISSUE TITLE
Cli hangs on errors

### ISSUE THREAD

**Issue by h4sh3d** [farcaster-project/farcaster-node#481]
When running twice the offer creation, the second should and do fail, an error is correctly display in node's log but the cli hangs waiting on a return.

Example of command
```
swap-cli make --btc-addr tb1q935eq5fl2a3ajpqp0e3d7z36g7vctcgv05f5lf\
    --xmr-addr 54EYTy2HYFcAXwAbFQ3HmAis8JLNmxRdTC9DwQL7sGJd4CAUYimPxuQHYkMNg1EELNP85YqFwqraLd4ovz6UeeekFLoCKiu\
    --btc-amount "0.0000135 BTC" --xmr-amount "0.001 XMR"\
    --network testnet --arb-blockchain bitcoin --acc-blockchain monero\
    --maker-role Bob --cancel-timelock 4 --punish-timelock 5 --fee-strategy "1 satoshi/vByte"
```
Node logs
```
[2022-06-08T10:02:24Z INFO  farcaster_node::config] Loading config file at: /home/joel/.farcaster/farcasterd.toml
[2022-06-08T10:02:24Z INFO  farcaster_node::service] New service farcasterd started
[2022-06-08T10:02:25Z INFO  farcaster_node::service] New service walletd started
[2022-06-08T10:02:25Z INFO  farcaster_node::farcasterd::runtime] Service walletd is now connected
[2022-06-08T10:02:29Z INFO  farcaster_node::farcasterd::runtime] Starting listener for incoming peer connections on 0.0.0.0:9735
[2022-06-08T10:02:29Z INFO  farcaster_node::farcasterd::runtime] Connection daemon listens for incoming peer connections on 0.0.0.0:9735
[2022-06-08T10:02:29Z INFO  farcaster_node::farcasterd::runtime] Public offer registered: 0x25ff…442c
[2022-06-08T10:02:29Z INFO  peerd] Local node id: 02de8fd55c1a4bec51ba245ecc814538d12af6665b175b13bc566920b705cd53ee
[2022-06-08T10:02:31Z INFO  farcaster_node::farcasterd::runtime] Connection daemon listens for incoming peer connections on 0.0.0.0:9735
[2022-06-08T10:02:31Z ERROR farcaster_node::farcasterd::runtime] This Public offer was previously registered
[2022-06-08T10:02:31Z ERROR microservices::esb::controller] ESB request processing error: This Public offer was previously registered
```

**Comment by TheCharlatan** [comment]
Works fine for me. I think this was fixed in #437 . Are you sure this was on latest `main`?

**Comment by h4sh3d** [comment]
This is no longer true, I close the issue.
